internal/config/cli.go

```diff
@@ -0,0 +1,163 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/spf13/cobra"
+)
+
+const ENV_VAR_PREFIX = "SIDECAR"
+
+// Options holds the configuration values from the command-line arguments and environment variables.
+type Options struct {
+	Network            string
+	Environment        string
+	Debug              bool
+	EthereumRPCBaseURL string
+	EthereumWSURL      string
+	StatsdUrl          string
+	SqliteInMemory     bool
+	SqliteDBFilePath   string
+	RPCGRPCPort        int
+	RPCHTTPPort        int
+	EtherscanAPIKeys   string
+}
+
+func ParseArgs(args []string, envs map[string]string) (*Config, error) {
+	opts, err := ParseArgsAndEnvironment(args, envs)
+	if err != nil {
+		return nil, err
+	}
+	if opts != nil {
+		return NewConfig(opts), nil
+	}
+	return nil, nil
+}
+
+// ParseArgs parses command-line arguments and environment variables.
+func ParseArgsAndEnvironment(args []string, envs map[string]string) (*Options, error) {
+	hasResult := false
+	opts := &Options{}
+
+	var rootCmd = &cobra.Command{
+		Use:   "sidecar",
+		Short: "Sidecar indexer",
+	}
+
+	var runCmd = &cobra.Command{
+		Use:   "run",
+		Short: "Run the sidecar indexer",
+
+		RunE: func(cmd *cobra.Command, cmdArgs []string) error {
+			// Validate required flags
+			if _, err := ParseNetwork(opts.Network); err != nil {
+				return fmt.Errorf("--network is required: %v", err)
+			}
+			if opts.Network == "mainnet" && opts.Environment == "" {
+				opts.Environment = "mainnet"
+			}
+			if _, err := parseEnvironment(opts.Environment); err != nil {
+				return fmt.Errorf("--environment is required: %v", err)
+			} else if opts.Network == "mainnet" && opts.Environment != "mainnet" {
+				return fmt.Errorf("mainnet environment required")
+			}
```

Comment:
One thing I mentioned in the original ticket was that we should do away with the network/environment pairing in favor of something like `--chain=<mainnet, holesky, preprod>`.

Environment/network has been a confusing concept for people using our other CLIs like the root updater and has really only been used internally, so I think a single flag to denote the target chain makes that much easier for people to understand.

Reply:
Yeah that does make sense!  This also is a nice simplification.  It is much more clear now.